### ISSUE TITLE
Class Riemannian KNN?

### ISSUE THREAD

**Issue by ninamiolane** [geomstats/geomstats#827]
@nguigs showed an example of Riemannian KNNs in our SciPy paper (see http://procbuild.scipy.org/download/geomstats-geomstats-paper).

This code does not seem to need the class that @YannCabanes implemented.

@YannCabanes, why did we need to wrap the KNN scikit-learn class here:
https://github.com/geomstats/geomstats/blob/master/geomstats/learning/knn.py ?

If we can just call scikit-learn's KNN, then we might not need to maintain a new class in geomstats?

**Comment by YannCabanes** [comment]
I did it because I wanted the knn classifier to be easier and faster to use for Geomstats users.
I mainly wanted to force the value of parameter 'algorithm' to be equal to 'brute'.
The other parameters of the scikit-learn knn classifier are not compatible with Riemannian metrics in general. 
I suppressed the parameter 'leaf_size' which is useless when the parameter 'algorithm' is set to 'brute'.
I also changed the name of the parameter 'metric' to 'distance' not the confuse Geomstats users: this parameter should be set to space.metric.dist and not space.metric
There might be another reason I do not remember that I wrote in the code reviews.

**Comment by nguigs** [comment]
I tried to use Yann's class but did not manage because
- there was stg wrong with the kwarg metric/distance ('got an unexpected keyword argument')
- sklearn checks that the imput array is 2d, so it doesn't work with matrices.
As I was in a rush I computed the pairwise distance matrix by hand and passed it to sklearn knn with metric='precomputed'

The sklearn class can be used, but there is a certain number of things to do to make it work, that's what Yann wanted to take care of for geomstats users, but it is not complete yet. We can either make it work with `point_type` args, or remve the class and leave the notebook + examples for people to see how to directly use the sklearn one

**Comment by ninamiolane** [comment]
Got it! Sorry @YannCabanes if I made you repeat something you already said in the PRs/slack! 

@nguigs wasn't the example helping wrt the use of KNNs?
https://github.com/geomstats/geomstats/blob/master/examples/plot_knn_s2.py

**Comment by nguigs** [comment]
> @nguigs wasn't the example helping wrt the use of KNNs?
> https://github.com/geomstats/geomstats/blob/master/examples/plot_knn_s2.py

Yes but the way it's done there doesn't work if `point_type='matrix'`
